backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/PredicatesHandler.java

```diff
@@ -105,25 +105,15 @@ private String getEntityName() {
     switch (key) {
       case "id":
       case "name":
-        return getContextFilterForSingleStringFields(predicate, key, value, bindingName);
+        return getContextFilterForSingleStringFields(
+            predicate, key, value.getStringValue(), bindingName);
       case "date_created":
-        var date =
-            value.getKindCase().equals(Value.KindCase.STRING_VALUE)
-                ? Double.parseDouble(value.getStringValue())
-                : value.getNumberValue();
-        return InternalFuture.completedInternalFuture(
-            new QueryFilterContext()
-                .addCondition(String.format("%s.date_created = :%s", alias, bindingName))
-                .addBind(q -> q.bind(bindingName, date)));
       case "date_updated":
-        var dateUpdated =
+        Object date =
```

Comment:
I really, really don't like this. Can we do something better with concrete types?

Reply:
Fixed